src/Features/Core/Portable/Completion/Providers/AbstractAwaitCompletionProvider.cs

```diff
@@ -43,41 +65,67 @@ public sealed override async Task ProvideCompletionsAsync(CompletionContext cont
+            var completionItem = GetCompletionItem(syntaxContext.TargetToken, isAwaitKeywordContext, isDotAwaitContext, generator, syntaxKinds, syntaxFacts);
             context.AddItem(completionItem);
         }
 
         public sealed override async Task<CompletionChange> GetChangeAsync(Document document, CompletionItem item, char? commitKey = null, CancellationToken cancellationToken = default)
         {
-            // IsComplexTextEdit is true when we want to add async to the container.
+            // IsComplexTextEdit is true when we want to add async to the container or place await in front of the expression.
             if (!item.IsComplexTextEdit)
             {
                 return await base.GetChangeAsync(document, item, commitKey, cancellationToken).ConfigureAwait(false);
             }
+            var completionChangeEdit = (CompletionChangeEdit)Enum.Parse(typeof(CompletionChangeEdit), item.Properties[nameof(CompletionChangeEdit)]);
+            using var _ = ArrayBuilder<TextChange>.GetInstance(out var builder);
+            var syntaxTree = await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var syntaxKinds = document.GetRequiredLanguageService<ISyntaxKindsService>();
 
-            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            var declaration = GetAsyncSupportingDeclaration(root.FindToken(item.Span.Start));
-            if (declaration is null)
+            if ((completionChangeEdit & CompletionChangeEdit.MakeContainerAsync) != 0)
             {
-                // IsComplexTextEdit should only be true when GetAsyncSupportingDeclaration returns non-null.
-                // This is ensured by the ShouldMakeContainerAsync overrides.
-                Debug.Assert(false, "Expected non-null value for declaration.");
-                return await base.GetChangeAsync(document, item, commitKey, cancellationToken).ConfigureAwait(false);
+                var root = await syntaxTree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+                var declaration = GetAsyncSupportingDeclaration(root.FindToken(item.Span.Start));
+                if (declaration is null)
+                {
+                    // IsComplexTextEdit should only be true when GetAsyncSupportingDeclaration returns non-null.
+                    // This is ensured by the ShouldMakeContainerAsync overrides.
+                    Debug.Assert(false, "Expected non-null value for declaration.");
+                    return await base.GetChangeAsync(document, item, commitKey, cancellationToken).ConfigureAwait(false);
+                }
+
+                builder.Add(new TextChange(new TextSpan(GetSpanStart(declaration), 0), syntaxFacts.GetText(syntaxKinds.AsyncKeyword) + " "));
```

Comment:
```suggestion
                builder.Add(new TextChange(new TextSpan(GetSpanStart(declaration), 0), "async "));
```

Reply:
I think this would break VB, because it's `Async` there.

Reply:
oh sorry, i didn't realize this was shared.  ignore this then :)